### ISSUE TITLE
MD046 mistakenly flags list item continuation as an indented code block

### ISSUE THREAD

**Issue by caugner** [DavidAnson/markdownlint#509]
[MD046](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md046---code-block-style) mistakenly flags the continuation of the list item in the following example as an indented code block.

# Example

## Source

```md
1. Simple item.
2. Item with code fence and continuation.

    ```shell
    git --version
    ```

    Continuation.
```

## Rendered

1. Simple item.
2. Item with code fence and continuation.

    ```shell
    git --version
    ```

    Continuation.

**Comment by DavidAnson** [comment]
I do not see this with your example, can you please check?

https://dlaa.me/markdownlint/#%25m%23%20Issue%20509%0A%0A1.%20Simple%20item.%0A2.%20Item%20with%20code%20fence%20and%20continuation.%0A%0A%20%20%20%20%60%60%60shell%0A%20%20%20%20git%20--version%0A%20%20%20%20%60%60%60%0A%0A%20%20%20%20Continuation.%0A

**Comment by caugner** [comment]
You are right, markdownlint works as intended here.

The problem was that the rest of the document (where I observed this) used indentation for code exclusively, and after adding this one code fence at the beginning, it was _actually_ reporting all following indented code blocks.

We could have avoided this situation somewhat ourselves by specifying the desired code block format in a markdownlint config. 🤷
